File: maincourse/libft_2/ft_strcpy.c
#include <string.h>
#include <stdio.h>

char    *ft_strcpy(char *s1, const char *s2)
{
    char *start_s1 = s1;

        while (*s2)
        {
            *s1 = *s2;
            s1++;
            s2++;
        }
        *s1 = '\0';
    return (start_s1);
}

int main()
{
    char source[] = "Hello, world!";
    char destination[13];

    ft_strcpy(destination, source);
    printf("Destination: %s\n", destination);

    return 0;
}

File: maincourse/libft_2/libft_py/ft_strcpy.py
def strncpy(dest, src, n):
    dest[:n] = src[:n]
    return dest

# Example usage
source = "Hello, world!"
destination = bytearray(13)  # A bytearray to hold the destination

# Copy at most 11 characters from source to destination
n = min(len(source), len(destination))
result = strncpy(destination, source.encode(), n)

# Convert the result back to a string for printing
result_str = result.decode()
print("Destination:", result_str)
